scylla/src/transport/cluster.rs

```diff
@@ -274,6 +278,7 @@ impl ClusterData {
         known_peers: &HashMap<Uuid, Arc<Node>>,
         used_keyspace: &Option<VerifiedKeyspaceName>,
         host_filter: Option<&dyn HostFilter>,
+        replication_opts: Option<&ReplicationConfigs>,
     ) -> Self {
```

Comment:
Let's not make it an `Option`; `ReplicationConfig` should be always passed to where `ReplicaLocator` is constructed.